float64[] positions
---
float64[] positions